xarray/core/pycompat.py

```diff
@@ -17,3 +17,11 @@
     sparse_array_type = (sparse.SparseArray,)
 except ImportError:  # pragma: no cover
     sparse_array_type = ()
+
+try:
+    # solely for isinstance checks
+    import cupy
+
+    cupy_array_type = (cupy.core.core.ndarray,)
```

Comment:
is there an additional `.core`?

Reply:
Sadly not :(. Although it looks like `cupy.ndarray` exists.